Fixed #33966 -- Added support for using KeyTextTransform from lookup. (#16016)
django/django: docs/topics/db/queries.txt

    >>> Dog.objects.filter(data__owner__isnull=True)
    <QuerySet [<Dog: Shep>]>

> .. class:: json.KT(lookup)

Reviewer: ```suggestion
``KT()`` expressions
~~~~~~~~~~~~~~~~~~~~

.. versionadded:: 4.2

.. class:: json.KT(lookup)
```

    A class that returns the resulting transformed JSON in the form of text.

>     ``lookup`` argument be a string starting with the name of the
>     :class:`~django.db.models.JSONField` and subsequent key or index transforms
>     separated by ``__`` (double underscore).

Reviewer: Maybe:
```suggestion
Represents the text value of a key, index, or path transform of
:class:`~django.db.models.JSONField`. You can use the double underscore
notation in `lookup` to chain dictionary keys and index transforms. For
example::
```

    To text transform :class:`~django.db.models.JSONField` - ``Dog.data``::

      >>> Dog.objects.create(name='Shep', data={'breed': ['collie', 'lhasa apso']})
      <Dog: Shep>
      >>> Dogs.objects.annotate(breed=KT("data__breed__1")).filter(breed='lhasa apso')

>       <QuerySet [<Dog: Shep>]>

Reviewer: I would add chain `KT` with a lookup and a second annotation, e.g.
```suggestion
    >>> Dog.objects.create(name="Shep", data={
    ...     "owner": {"name": "Bob"},
    ...     "breed": ["collie", "lhasa apso"],
    ... })
    <Dog: Shep>
    >>> Dogs.objects.annotate(
    ...     first_breed=KT("data__breed__1"),
    ...     owner=KT("data__owner__name"),
    ... ).filter(breed__startswith="lhasa", owner="Bob")
    <QuerySet [<Dog: Shep>]>
